util/ValueSetHelper.ts

```diff
@@ -0,0 +1,179 @@
+
+          if (primaryCodeInfo.codeType === 'FHIR.CodeableConcept') {
+            if (primaryCodeInfo.choiceType === true) {
+              if (primaryCodeInfo.multipleCardinality === true) {
+                // not sure if this happens based on codePaths.ts
+              } else {
+                // example: MedicationRequest, DeviceRequest
+                const primaryCodeValue = fhirpath.evaluate(
+                  r.resource,
+                  `${codeInfo.primaryCodePath}CodeableConcept`
+                )[0] as fhir4.CodeableConcept;
+                checkCodesAndValueSets(primaryCodeValue, matchingDRType, measureBundle, r, newResources);
+              }
+            } else {
+              if (primaryCodeInfo.multipleCardinality === true) {
+                // example: Activity Definition, Appointment, Encounter
+                const primaryCodeValue = fhirpath.evaluate(
+                  r.resource,
+                  codeInfo.primaryCodePath
+                ) as fhir4.CodeableConcept[];
+                if (primaryCodeValue) {
+                  primaryCodeValue.forEach(pcv => {
+                    checkCodesAndValueSets(pcv, matchingDRType, measureBundle, r, newResources);
+                  });
+                }
+              } else {
+                const primaryCodeValue = fhirpath.evaluate(
+                  r.resource,
+                  codeInfo.primaryCodePath
+                )[0] as fhir4.CodeableConcept;
+                checkCodesAndValueSets(primaryCodeValue, matchingDRType, measureBundle, r, newResources);
+              }
+            }
+          } else if (primaryCodeInfo.codeType === 'FHIR.Coding') {
+            if (primaryCodeInfo.choiceType === true) {
+              if (primaryCodeInfo.multipleCardinality === true) {
+                // not sure if this happens based on codePaths.ts
+              } else {
+                // example: MessageDefinition
+                const primaryCodeValue = fhirpath.evaluate(
+                  r.resource,
+                  `${codeInfo.primaryCodePath}Coding`
+                ) as fhir4.Coding;
+                if (primaryCodeValue) {
+                  if (
+                    matchingDRType.directCodes.length > 0 &&
+                    matchingDRType.directCodes.find(
+                      dc => dc.code === primaryCodeValue.code && dc.system === primaryCodeValue.system
+                    )
+                  ) {
+                    newResources.push(r);
+                  } else if (matchingDRType.valueSets.length > 0) {
+                    const vsCodesAndSystems = getValueSetCodes(matchingDRType.valueSets, measureBundle);
+                    if (
+                      vsCodesAndSystems.find(
+                        vscas => primaryCodeValue.code === vscas.code && primaryCodeValue.system === vscas.system
+                      )
+                    ) {
+                      newResources.push(r);
+                    }
+                  }
```

Comment:
This looks like it is the same code in the `checkCodesAndValueSets` function.

Reply:
It's a little different